test/model/store_test.dart

```diff
@@ -427,6 +429,36 @@ void main() {
       check(store.userSettings!.twentyFourHourTime).isTrue();
     }));
 
+    test('expired queue disposes registered MessageListView instances', () => awaitFakeAsync((async) async {
+      // Regression test for: https://github.com/zulip/zulip-flutter/issues/810
+      await prepareStore();
+      updateMachine.debugPauseLoop();
+      updateMachine.poll();
+      check(globalStore.perAccountSync(store.accountId)).identicalTo(store);
+
+      // Make sure there are message list views in the message store.
+      MessageListView.init(store: store, narrow: const MentionsNarrow());
+      MessageListView.init(store: store, narrow: const StarredMessagesNarrow());
+      check(store.debugMessageStore.debugMessageLists).length.equals(2);
+
+      // Let the server expire the event queue.
+      connection.prepare(httpStatus: 400, json: {
+        'result': 'error', 'code': 'BAD_EVENT_QUEUE_ID',
+        'queue_id': updateMachine.queueId,
+        'msg': 'Bad event queue ID: ${updateMachine.queueId}',
+      });
+      updateMachine.debugAdvanceLoop();
+      async.flushMicrotasks();
+      await Future<void>.delayed(Duration.zero);
+      check(store).isLoading.isTrue();
+      check(store.debugMessageStore.debugMessageLists).isEmpty();
+
+      // The global store has a new store.
+      check(globalStore.perAccountSync(store.accountId)).not((it) => it.identicalTo(store));
+      updateFromGlobalStore();
+      check(store).isLoading.isFalse();
```

Comment:
This can be more focused on the aspect this particular test is about:
```suggestion
      await Future<void>.delayed(Duration.zero);

      // The old store's [MessageListView]s have been disposed.
      // (And no exception was thrown; that was #810.)
      check(store.debugMessageStore.debugMessageListViews).isEmpty();
```

The other checks are effectively covered by the preceding test case.

Meanwhile the explicit comment about no exception is helpful because in this test case that's the main point of the test. Every test case implicitly checks that no miscellaneous exceptions were thrown from the code under test; but usually that's not the main thing the test is about, so when it's not mentioned the reader might overlook it.